### PR TITLE
Improve phy rtl8211f driver

### PR DESCRIPTION
**Pull request by yangbolu1991**
During using current rtl8211 phy driver on i.MX95 EVK board, two issues were found.
1. Without gpio interrupt used, the driver couldn't detect link change since the monitor work hadn't been scheduled.
2. The tx/rx failed because of RGMII timing issue. The TX/RX delay should be enabled to ensure RGMII timing.

The two patches fixed the issues.

Thanks.

### CODE REVIEW

**Comment by JiafeiPan** [comment]
Please fix compliance checking warnings.

**Review by pdgendt — COMMENTED** [review]
Commit body isn't clear to me what has been changed, also can you fill in a PR subject/description that reflects the change?

> Enable RGMII TX/RX delay to ensure timing.

**Comment by yangbolu1991** [comment]
> Commit body isn't clear to me what has been changed, also can you fill in a PR subject/description that reflects the change?
> 
> > Enable RGMII TX/RX delay to ensure timing.

Thanks @pdgendt . I updated the commit message and PR information to make it clear:)

**Review by pdgendt — DISMISSED** [review]

File: drivers/ethernet/phy/phy_realtek_rtl8211f.c [hunk 1]
```diff
@@ -589,6 +634,8 @@ static int phy_rt_rtl8211f_init(const struct device *dev)
 		return ret;
 	}
 skip_int_gpio:
+#else
+	phy_rt_rtl8211f_monitor_work_handler(&data->phy_monitor_work.work);
```
Comment (stanislav-poboril) [inline drivers/ethernet/phy/phy_realtek_rtl8211f.c:638]: Polling is sometimes done even if `DT_ANY_INST_HAS_PROP_STATUS_OKAY(int_gpios)` is true:
In a hypothetical situation when there are multiple RTL8211F PHYs, and one has `int-gpios` property set, but the actual instance does not and uses polling. So we need to look if the port is set for the instance, similar to what other parts of this source file do:
```
#if DT_ANY_INST_HAS_PROP_STATUS_OKAY(int_gpios)
	if (!config->interrupt_gpio.port) {
		/* also start monitoring */
	}
#else
```

Comment (yangbolu1991) [reply drivers/ethernet/phy/phy_realtek_rtl8211f.c:639]: > Polling is sometimes done even if `DT_ANY_INST_HAS_PROP_STATUS_OKAY(int_gpios)` is true: In a hypothetical situation when there are multiple RTL8211F PHYs, and one has `int-gpios` property set, but the actual instance does not and uses polling. So we need to look if the port is set for the instance, similar to what other parts of this source file do:
> 
> ```
> #if DT_ANY_INST_HAS_PROP_STATUS_OKAY(int_gpios)
> 	if (!config->interrupt_gpio.port) {
> 		/* also start monitoring */
> 	}
> #else
> ```

Thanks for the suggestion. I updated a new version. Only need to schedule work for the first time in init. The handling of other parts seems okay.

**Review by stanislav-poboril — APPROVED** [review]

**Review by JiafeiPan — APPROVED** [review]

**Review by pdgendt — APPROVED** [review]

**Review by stanislav-poboril — APPROVED** [review]

**Pull request merged**